src/alchemlyb/tests/test_preprocessing.py

```diff
@@ -68,6 +68,29 @@ def slicer(self, *args, **kwargs):
     def test_basic_slicing(self, data, size):
         assert len(self.slicer(data, lower=1000, upper=34000, step=5)) == size
 
+    @pytest.mark.parametrize(('data', 'lower', 'upper'),
+                             [
+                                 (gmx_benzene_dHdl(), 1000, 34000),
+                                 (gmx_benzene_u_nk(), 1000, 34000),
+                             ])
+    def test_lower_and_upper_bound(self, data, lower, upper):
+        """
+        Test that the lower and upper time is respected
+        """
+        original_length = len(data)
+        # Check that the input data is appropriate for the test
+        assert any(data.reset_index()['time'] < lower)
+        assert any(data.reset_index()['time'] > upper)
+
+        # Slice data, and check that we don't observe times outside
+        # the prescribed range
+        sliced = self.slicer(data, lower=lower, upper=upper, step=5)
+        assert all(sliced.reset_index()['time'] >= lower)
+        assert all(sliced.reset_index()['time'] <= upper)
+
+        # Make sure we didn't change input data
+        assert len(data) == original_length
```

Comment:
Why do you need to check if the input data changes in the unit test?

Reply:
I think it's valuable to check that slicing does not change the input data. However, this might be cleaner as its own unit test.

Shall I implement this? Essentially, I would duplicate the entire test until the call to the slicer, then assert only that the original data is unchanged, and call it something like `test_preserves_original_data`.

Or, we could change the name of the current test to `test_lower_and_upper_bound_and_preserves_original_data` to reflect that we are testing two things.

Which approach would you suggest?

Reply:
Separate test would be cleaner, especially once it fails. I think it's sufficient to check this with just a single input data set. You probably also don't need the check that the data are appropriate for the test.

Reply:
Extracted this into separate test.